lib/evaluation/handlers/limitDisclosureEvaluationHandler.ts

```diff
@@ -19,18 +20,39 @@ export class LimitDisclosureEvaluationHandler extends AbstractEvaluationHandler
 
   public handle(pd: PresentationDefinition, vcs: VerifiableCredential[]): void {
     pd.input_descriptors.forEach((inDesc: InputDescriptor, index: number) => {
-      if (inDesc.constraints?.fields && inDesc.constraints?.limit_disclosure === Optionality.Required) {
-        this.enforceLimitDisclosure(vcs, inDesc.constraints.fields, index);
+      if (
+        inDesc.constraints?.fields &&
+        (inDesc.constraints?.limit_disclosure === Optionality.Required ||
```

Comment:
In case of optional it should really look at the proof of the vc to determine whether selective disclosure needs to be applied or not

Reply:
It already looks at the proof of the vc.

Reply:
It does that in the wrong location, or the name of that method is wrong (enforce)

Reply:
Got it! The name of the method is wrong. Fixed!